pkg/hsskey/filereader.go

```diff
@@ -156,3 +159,75 @@ func getHssFromFile(verbose bool, verboseDangerous bool, filePaths []string) ([]
 
 	return allHss, nil
 }
+
+// WriteHssToTempFile writes a list of HSS to a tmpfs file where the filepath is returned.
+// See WriteHssToFile for HSS details.
+func WriteHssToTempFile(verbose bool, verboseDangerous bool, hss [][]byte) (string, error) {
+	file, err := os.CreateTemp("", "hss_*")
+	if err != nil {
+		return "", fmt.Errorf("failed to create temp file %v", err)
+	}
+
+	err = WriteHssToFile(verbose, verboseDangerous, file, hss)
+	if err != nil {
+		file.Close()
+		os.Remove(file.Name())
+		return "", err
+	}
+
+	file.Close()
+	return file.Name(), nil
+}
+
+// WriteHssToFile writes a list of HSS to an open file.
+// Each HSS key is expected to be 64 bytes long and has a checksum for validation.
+// HSS that fail to validate are not written to the file.
+// Function returns error if no HSS are written or writing can no longer continue.
+func WriteHssToFile(verbose bool, verboseDangerous bool, file *os.File, hss [][]byte) error {
+	if len(hss) == 0 {
+		return fmt.Errorf("InvalidArgument: No HSS to write")
+	}
+	if file == nil {
+		return fmt.Errorf("InvalidArgument: file is nil")
+	}
+	// Count of successful writes
+	written := 0
+
+	for i := 0; i < len(hss); i++ {
+		hssKey := hss[i]
+
+		// Verify the checksums.
+		if !validateChecksum(hssKey) {
+			msg := fmt.Sprintf("Checksum validation failed for HSS key index %d", i)
+			if verboseDangerous {
+				msg = msg + fmt.Sprintf("\nseed=%x, seed(octal escape sequence)=%s", hssKey,
+					toOctalEscapeSequence(hssKey))
+			}
+			log.Print(msg)
+			continue
+		}
+
+		if verbose {
+			msg := fmt.Sprintf("Writing HSS key to file=%s", file.Name())
+			if verboseDangerous {
```

Comment:
for testing, it is useful to pass an io.Writer to functions such as this, so that error output can be managed. Also, having these log.Print all over the package makes it very hard to read test output when things fail. Can you change this?

Reply:
cmiaw, is verbose flags really passed in for unit/automated testing?
I would imagine the verbose/verboseDangerous flags already provide sufficient levers to manage test output during debugging.

I do see log.Printfs in 50+ other pkg files

Reply:
yes, those crept in as people moved command code to packages. 
Just because it is done does not make it right. Let's please not add more.

Reply:
and, yes, you want to pass in all value of all flags, that's how you make sure tests cover all cases.

Reply:
Added tests here to pass in flag combinations for io.Writer + dangerious bool.